src/rpm/signature/pgp.rs

```diff
@@ -106,16 +111,22 @@ impl Verifier {
 
 impl traits::Verifying<traits::algorithm::RSA> for Verifier {
     type Signature = Vec<u8>;
-    fn verify(&self, data: &[u8], signature: &[u8]) -> Result<(), RPMError> {
+    fn verify<R: Read>(&self, mut data: R, signature: &[u8]) -> Result<(), RPMError> {
         let signature = Self::parse_signature(signature)?;
 
         log::debug!("Signature issued by: {:?}", signature.issuer());
 
+        // currently the copy is necessary due to a lack of API allowing
+        // something `std::io::Read` based: https://github.com/rpgp/rpgp/issues/99
+        let mut buf = Vec::with_capacity(16384);
```

Comment:
I think this behaviour should be documented in rustdoc for this struct since the API now gives the false impression that there is no copy involved. Just as a fair disclaimer.